Algebra plugins configurations not working as intended

So here's an interesting little CMake problem. I'm trying to compile detray against a system installation of the algebra plugins, which is giving me the following configuration-time error:

```
CMake Error at /lib/cmake/Vc/VcConfig.cmake:11 (message):
  File or directory
  /lib/cmake/algebra-plugins-0.2
  referenced by variable algebra_plugins_CMAKE_DIR does not exist !
Call Stack (most recent call first):
  /lib/cmake/algebra-plugins-0.2/algebra-plugins-config.cmake:51 (set_and_check)
  CMakeLists.txt:67 (find_package)
```

I'm suspecting that this is a bug in the algebra plugins build system, specifically in `/cmake/algebra-plugins-config.cmake.in`. I don't really understand why the Vc configuration is trying to look for the algebra plugins here, that seems like the opposite of what should be happening.

The buggy line in question (`/lib/cmake/algebra-plugins-0.2/algebra-plugins-config.cmake:51`) is the following:

```cmake
set_and_check( algebra_plugins_CMAKE_DIR "${PACKAGE_PREFIX_DIR}/lib/cmake/algebra-plugins-0.2" )
```

Maybe @krasznaa has any ideas on what might be going on?

Re-opening this issue because the problem is still here, and it's giving me grief still. Thankfully this time I have a clear reproducer in the form of a Dockerfile:

```dockerfile
FROM ghcr.io/acts-project/ubuntu2004_cuda:v27

# DEBIAN_FRONTEND ensures non-blocking operation (tzdata is a problem)
ENV DEBIAN_FRONTEND noninteractive

# Install dfelibs
RUN git clone https://github.com/acts-project/dfelibs.git --branch v20211029 /tmp/dfelibs/
RUN cmake -B /tmp/dfelibs_build/ -S /tmp/dfelibs/ -DCMAKE_INSTALL_PREFIX=/
RUN cmake --build /tmp/dfelibs_build/ -- -j $(nproc)
RUN cmake --install /tmp/dfelibs_build/

# Install vecmem
RUN git clone https://github.com/acts-project/vecmem.git --branch v0.17.0 /tmp/vecmem/
RUN cmake -B /tmp/vecmem_build/ -S /tmp/vecmem/ \
    -DCMAKE_INSTALL_PREFIX=/ \
    -DVECMEM_BUILD_TESTING=Off
RUN cmake --build /tmp/vecmem_build/ -- -j $(nproc)
RUN cmake --install /tmp/vecmem_build/

# Install Vc
RUN git clone https://github.com/VcDevel/Vc.git --branch 1.4.2 /tmp/vc/
RUN cmake -B /tmp/vc_build/ -S /tmp/vc/ \
    -DCMAKE_INSTALL_PREFIX=/
RUN cmake --build /tmp/vc_build/ -- -j $(nproc)
RUN cmake --install /tmp/vc_build/

# Install Algebra Plugins
RUN git clone https://github.com/acts-project/algebra-plugins.git --branch v0.10.0 /tmp/algebra-plugins/
RUN cmake -B /tmp/algebra-plugins_build/ -S /tmp/algebra-plugins/ \
    -DCMAKE_INSTALL_PREFIX=/ \
    -DALGEBRA_PLUGINS_BUILD_TESTING=Off \
    -DALGEBRA_PLUGINS_SETUP_GOOGLETEST=Off \
    -DALGEBRA_PLUGINS_USE_SYSTEM_EIGEN3=On \
    -DALGEBRA_PLUGINS_USE_SYSTEM_VC=On \
    -DALGEBRA_PLUGINS_USE_SYSTEM_VECMEM=On \
    -DALGEBRA_PLUGINS_INCLUDE_EIGEN=On \
    -DALGEBRA_PLUGINS_INCLUDE_VC=On
RUN cmake --build /tmp/algebra-plugins_build/ -- -j $(nproc)
RUN cmake --install /tmp/algebra-plugins_build/

# Install Detray
RUN git clone https://github.com/acts-project/detray.git --branch v0.11.0 /tmp/detray/
RUN cmake -B /tmp/detray_build/ -S /tmp/detray/ \
    -DCMAKE_INSTALL_PREFIX=/ \
    -DDETRAY_BUILD_TESTING=Off \
    -DDETRAY_BENCHMARKS=Off \
    -DDETRAY_SETUP_BENCHMARK=Off \
    -DDETRAY_USE_SYSTEM_DFELIBS=On \
    -DDETRAY_USE_SYSTEM_THRUST=On \
    -DDETRAY_USE_SYSTEM_GOOGLETEST=On \
    -DDETRAY_USE_SYSTEM_BENCHMARK=On \
    -DDETRAY_USE_SYSTEM_VECMEM=On \
    -DDETRAY_USE_SYSTEM_ALGEBRA_PLUGINS=On \
    -DDETRAY_USE_SYSTEM_EIGEN3=On
RUN cmake --build /tmp/detray_build/ -- -j $(nproc)
RUN cmake --install /tmp/detray_build/
```

The error is exactly the same as before:

```
CMake Error at /usr/lib/cmake/Vc/VcConfig.cmake:11 (message):
  File or directory /usr/usr/include referenced by variable
  algebra_plugins_INCLUDE_DIR does not exist !
Call Stack (most recent call first):
  /usr/lib/cmake/algebra-plugins-0.7/algebra-plugins-config.cmake:48 (set_and_check)
  CMakeLists.txt:88 (find_package)
```

Comments appreciated!

--- a/cmake/algebra-plugins-config.cmake.in
+++ b/cmake/algebra-plugins-config.cmake.in
@@ -1,6 +1,6 @@
 # Algebra plugins library, part of the ACTS project (R&D line)
 #
-# (c) 2021-2022 CERN for the benefit of the ACTS project
+# (c) 2021-2023 CERN for the benefit of the ACTS project
 #
 # Mozilla Public License Version 2.0
 
@@ -13,6 +13,12 @@ set( ALGEBRA_PLUGINS_INCLUDE_EIGEN @ALGEBRA_PLUGINS_INCLUDE_EIGEN@ )
 set( ALGEBRA_PLUGINS_INCLUDE_SMATRIX @ALGEBRA_PLUGINS_INCLUDE_SMATRIX@ )
 set( ALGEBRA_PLUGINS_INCLUDE_VC @ALGEBRA_PLUGINS_INCLUDE_VC@ )
 set( ALGEBRA_PLUGINS_INCLUDE_VECMEM @ALGEBRA_PLUGINS_INCLUDE_VECMEM@ )
+
+# Set up some simple variables for using the package.
+set( algebra_plugins_VERSION "@PROJECT_VERSION@" )
+set_and_check( algebra_plugins_INCLUDE_DIR
+   "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@" )
+set_and_check( algebra_plugins_CMAKE_DIR "@PACKAGE_CMAKE_INSTALL_CMAKEDIR@" )
 
 # Find all packages that Algebra Plugins needs to function.
 include( CMakeFindDependencyMacro )
@@ -28,12 +34,6 @@ endif()
 if( ALGEBRA_PLUGINS_INCLUDE_VECMEM )
    find_dependency( vecmem )
 endif()
-
-# Set up some simple variables for using the package.
-set( algebra_plugins_VERSION "@PROJECT_VERSION@" )
-set_and_check( algebra_plugins_INCLUDE_DIR
-   "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@" )
-set_and_check( algebra_plugins_CMAKE_DIR "@PACKAGE_CMAKE_INSTALL_CMAKEDIR@" )
 
 # Print a standard information message about the package being found.
 include( FindPackageHandleStandardArgs )

